Libraries/ART/ARTShape.m

```diff
@@ -28,7 +28,7 @@ - (void)dealloc
 
 - (void)renderLayerTo:(CGContextRef)context
 {
-  if ((!self.fill && !self.stroke) || !self.d) {
+  if ((!self.fill && !self.stroke && !self.gradientStroke.count) || !self.d) {
```

Comment:
Hi, `self.gradientStroke` cannot be null because I initialized it as an [empty array](https://github.com/facebook/react-native/pull/10802/files#diff-da10f4a773d6b8638a9442113994dd20R379)

So the condition `if ((!self.fill && !self.stroke && !self.gradientStroke.count) || !self.d)`  is used to check if Shape is provided with any prop `fill`, `stroke`, or `gradientStroke` or not at all, if no `fill`, `stroke`, or `gradientStroke` is provided or no `path` (which is `d` ) is provided, Shape component will stop drawing.

Therefore, I used `self.gradientStroke.count` to check if `gradientStroke` prop is empty or not because `self.gradientStroke.count` == 0 means the colors array in `gradientStroke` is empty.

Thanks for asking!